fields/types/password/PasswordType.js

```diff
@@ -123,10 +123,17 @@ password.prototype.validateInput = function (data, callback) {
 	var result = true;
 	var confirmValue = this.getValueFromData(data, '_confirm');
 	var passwordValue = this.getValueFromData(data);
-	if (passwordValue !== undefined && confirmValue !== undefined && passwordValue !== confirmValue) {
+	if (passwordValue !== undefined
+		&& confirmValue !== undefined
+		&& passwordValue !== confirmValue) {
 		result = false;
 		detail = 'passwords must match';
 	}
+	if ((passwordValue === undefined || confirmValue === undefined)
```

Comment:
Good catch. If the `password` is undefined but a `confirmValue` was passed, I think we should fail validation as if they were both provided and didn't match.